addons/html_builder/static/src/plugins/carousel_option_plugin.js

```diff
@@ -158,6 +159,63 @@ export class CarouselOptionPlugin extends Plugin {
             }
         });
     }
+
+    reorderCarouselItems({ elementToReorder, position, optionName }) {
+        if (optionName === "Carousel") {
+            const editingCarouselElement = elementToReorder.closest(".s_carousel");
+            const itemsEls = [...editingCarouselElement.querySelectorAll(".carousel-item")];
+
+            // reorder carousel items
+            const oldPosition = itemsEls.indexOf(elementToReorder);
+            if (oldPosition === 0 && position === "prev") {
+                position = "last";
+            } else if (oldPosition === itemsEls.length - 1 && position === "next") {
+                position = "first";
+            }
+            itemsEls.splice(oldPosition, 1);
+            switch (position) {
+                case "first":
+                    itemsEls.unshift(elementToReorder);
+                    break;
+                case "prev":
+                    itemsEls.splice(Math.max(oldPosition - 1, 0), 0, elementToReorder);
+                    break;
+                case "next":
+                    itemsEls.splice(oldPosition + 1, 0, elementToReorder);
+                    break;
+                case "last":
+                    itemsEls.push(elementToReorder);
+                    break;
+            }
+
+            // replace the carousel-inner element by one with reordered carousel items
+            const carouselInnerEl = editingCarouselElement.querySelector(".carousel-inner");
+            const newCarouselInnerEl = document.createElement("div");
+            newCarouselInnerEl.classList.add("carousel-inner");
+            newCarouselInnerEl.append(...itemsEls);
+            carouselInnerEl.replaceWith(newCarouselInnerEl);
+
+            // slide to the reordered target carousel item and update indicators
+            const newItemPosition = itemsEls.indexOf(elementToReorder);
+            editingCarouselElement.classList.remove("slide");
+            const carouselInstance = window.Carousel.getOrCreateInstance(editingCarouselElement, {
+                ride: false,
+                pause: true,
+            });
+            carouselInstance.to(newItemPosition);
+            const indicatorEls = editingCarouselElement.querySelectorAll(
+                ".carousel-indicators > *"
+            );
+            indicatorEls.forEach((indicatorEl, i) => {
+                indicatorEl.classList.toggle("active", i === newItemPosition);
+            });
+            const activeImageEl = editingCarouselElement.querySelector(".carousel-item.active img");
+            this.dependencies["builder-options"].updateContainers(activeImageEl);
+            editingCarouselElement.classList.add("slide");
+            // Prevent the carousel from automatically sliding afterwards.
+            carouselInstance["pause"]();
```

Comment:
you should do 
            this.dependencies["builder-options"].updateContainers(activeImageEl);
at the end and add a step just before (you need it for the undo)